startblog/views.py

```diff
@@ -54,81 +80,109 @@ def blog_update(request,pk):
-					"uid": urlsafe_base64_encode(force_bytes(user.pk)),
-					"user": user,
-					'token': default_token_generator.make_token(user),
-					'protocol': 'http',
-					}
-					email = render_to_string(email_template_name, c)
-					try:
-						send_mail(subject, email, 'admim@gmail.com' , [user.email], fail_silently=False)
-					except BadHeaderError:
-						return HttpResponse('Invalid header found.')
-					return redirect ("/password_reset/done/")
-	password_reset_form = PasswordResetForm()
-	return render(request=request, template_name="passwords/password_reset.html", context={"password_reset_form":password_reset_form})
+    if request.method == "POST": 
+        password_reset_form = PasswordResetForm(request.POST)
+        if password_reset_form.is_valid():
+            data = password_reset_form.cleaned_data['email']
+            associated_users = User.objects.filter(Q(email=data))
+            if associated_users.exists():
+                for user in associated_users:
+                    subject = "Password Reset Requested"
+                    email_template_name = "passwords/password_reset_email.txt"
+                    c = {
+                    "email":user.email,
+                    'domain':'127.0.0.1:8000',
+                    'site_name': 'Website',
+                    "uid": urlsafe_base64_encode(force_bytes(user.pk)),
+                    "user": user,
+                    'token': default_token_generator.make_token(user),
+                    'protocol': 'http',
+                    }
+                    email = render_to_string(email_template_name, c)
+                    try:
+                        '''delay is used as its a celery function in tasks.py '''
+                        send_mail_task.delay(subject,email,'sender@gmail.com',user.email)
+                    except BadHeaderError:
+                        return HttpResponse('Invalid header found.')
+                    return redirect ("/password_reset/done/")
+    password_reset_form = PasswordResetForm()
+    return render(request=request, template_name="passwords/password_reset.html", context={"password_reset_form":password_reset_form})
+
 
 def add_like(request,pk):
     blog = Blogs.objects.get(pk=pk)  
     if request.method =='POST':
-        if request.POST.get('like')=='like':
-            like =Likes.objects.filter(post=blog,like_status=False,liked_user=request.user)   
+        if request.POST.get('like') ==' like':
+            like =Likes.objects.filter(post=blog, like_status=False, liked_user=request.user)   
             if like:
                 like.update(like_status = True)
             else:
-                like = Likes.objects.create(post=blog,like_status=True,liked_user=request.user)
+                like = Likes.objects.create(post=blog, like_status=True, liked_user=request.user)
                 return redirect('startblog:home')
-        if request.POST.get('like')=='unlike':
+        if request.POST.get('like') == 'unlike':
```

Comment:
Use DeleteView for unlike and CreateView for Like

Reply:
Done